NoClassDefFoundError after submitting local index task

I was getting ```java.lang.NoClassDefFoundError: org/apache/hadoop/fs/FSDataInputStream``` after submitting a local wikipedia index task(same one as in the integration tests). I was using the latest code in the master branch.

Here is the stack trace,
```
2015-07-22T19:36:26,175 ERROR [task-runner-0] io.druid.indexing.overlord.ThreadPoolTaskRunner - Uncaught Throwable while running task[IndexTask{id=index_wikipedia_index_test_2015-07-22T19:36:21.081Z, type=index, dataSource=wikipedia_index_test}]
java.lang.NoClassDefFoundError: org/apache/hadoop/fs/FSDataInputStream
	at io.druid.indexing.common.task.IndexTask.determinePartitions(IndexTask.java:269) ~[classes/:?]
	at io.druid.indexing.common.task.IndexTask.run(IndexTask.java:193) ~[classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:235) [classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:214) [classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [?:1.7.0_75]
	at java.lang.Thread.run(Thread.java:745) [?:1.7.0_75]
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.fs.FSDataInputStream
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366) ~[?:1.7.0_75]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[?:1.7.0_75]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.7.0_75]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[?:1.7.0_75]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[?:1.7.0_75]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[?:1.7.0_75]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[?:1.7.0_75]
	... 8 more
2015-07-22T19:36:26,183 ERROR [main] io.druid.cli.CliPeon - Error when starting up.  Failing.
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/apache/hadoop/fs/FSDataInputStream
	at com.google.common.base.Throwables.propagate(Throwables.java:160) ~[guava-16.0.1.jar:?]
	at io.druid.indexing.worker.executor.ExecutorLifecycle.join(ExecutorLifecycle.java:162) ~[classes/:?]
	at io.druid.cli.CliPeon.run(CliPeon.java:225) [classes/:?]
	at io.druid.cli.Main.main(Main.java:91) [classes/:?]
Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/apache/hadoop/fs/FSDataInputStream
	at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:299) ~[guava-16.0.1.jar:?]
	at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:286) ~[guava-16.0.1.jar:?]
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116) ~[guava-16.0.1.jar:?]
	at io.druid.indexing.worker.executor.ExecutorLifecycle.join(ExecutorLifecycle.java:159) ~[classes/:?]
	... 2 more
Caused by: java.lang.NoClassDefFoundError: org/apache/hadoop/fs/FSDataInputStream
	at io.druid.indexing.common.task.IndexTask.determinePartitions(IndexTask.java:269) ~[classes/:?]
	at io.druid.indexing.common.task.IndexTask.run(IndexTask.java:193) ~[classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:235) ~[classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:214) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) ~[?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) ~[?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) ~[?:1.7.0_75]
	at java.lang.Thread.run(Thread.java:745) ~[?:1.7.0_75]
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.fs.FSDataInputStream
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366) ~[?:1.7.0_75]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[?:1.7.0_75]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.7.0_75]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[?:1.7.0_75]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[?:1.7.0_75]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[?:1.7.0_75]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[?:1.7.0_75]
	at io.druid.indexing.common.task.IndexTask.determinePartitions(IndexTask.java:269) ~[classes/:?]
	at io.druid.indexing.common.task.IndexTask.run(IndexTask.java:193) ~[classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:235) ~[classes/:?]
	at io.druid.indexing.overlord.ThreadPoolTaskRunner$ThreadPoolTaskRunnerCallable.call(ThreadPoolTaskRunner.java:214) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) ~[?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) ~[?:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) ~[?:1.7.0_75]
	at java.lang.Thread.run(Thread.java:745) ~[?:1.7.0_75]
2015-07-22T19:36:26,184 INFO [Thread-42] io.druid.cli.CliPeon - Running shutdown hook
2015-07-22 19:36:26,207 FATAL Unable to register shutdown hook because JVM is shutting down.
```

It looks like ```HadoopDruidIndexerConfig``` triggers classloader to load ```FSDataInputStream```. 
Specifically this line in ```IndexTask```, 
```
hashFunction.hashBytes(HadoopDruidIndexerConfig.jsonMapper.writeValueAsBytes(groupKey))
                          .asBytes()
```

--- a/indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java
+++ b/indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java
@@ -39,7 +39,6 @@ import io.druid.data.input.FirehoseFactory;
 import io.druid.data.input.InputRow;
 import io.druid.data.input.Rows;
 import io.druid.granularity.QueryGranularity;
-import io.druid.indexer.HadoopDruidIndexerConfig;
 import io.druid.indexing.common.TaskLock;
 import io.druid.indexing.common.TaskStatus;
 import io.druid.indexing.common.TaskToolbox;
@@ -267,8 +266,7 @@ public class IndexTask extends AbstractFixedIntervalTask
               inputRow
           );
           collector.add(
-              hashFunction.hashBytes(HadoopDruidIndexerConfig.jsonMapper.writeValueAsBytes(groupKey))
-                          .asBytes()
+              hashFunction.hashBytes(jsonMapper.writeValueAsBytes(groupKey)).asBytes()
           );
         }
       }
@@ -290,13 +288,7 @@ public class IndexTask extends AbstractFixedIntervalTask
       shardSpecs.add(new NoneShardSpec());
     } else {
       for (int i = 0; i < numberOfShards; ++i) {
-        shardSpecs.add(
-            new HashBasedNumberedShardSpec(
-                i,
-                numberOfShards,
-                HadoopDruidIndexerConfig.jsonMapper
-            )
-        );
+        shardSpecs.add(new HashBasedNumberedShardSpec(i, numberOfShards, jsonMapper));
       }
     }
 

